python/pyarrow/table.pyx

```diff
@@ -296,6 +296,56 @@ cdef class RecordBatch:
         return result
 
 
+cdef class RecordBatchList:
+    '''
+    A utility class used for conversions with RecordBatch lists
+    '''
+
+    @staticmethod
+    def to_pandas(batches):
```

Comment:
This is a Java-like construct -- just make this a plain old function "dataframe_from_batches" or something like that